tests/qemu-tests/conftest.py

```diff
@@ -0,0 +1,105 @@
+"""
+This file should consist of global test fixtures.
+"""
+
+from __future__ import annotations
+
+import os
+import subprocess
+import sys
+
+import gdb
+import pytest
+from pwn import context
+from pwn import make_elf_from_assembly
+
+_start_binary_called = False
+
+QEMU_PORT = os.environ.get("QEMU_PORT")
+
+
+@pytest.fixture
+def qemu_assembly_run():
+    """
+    Returns function that launches given binary with 'starti' command
+
+    The `path` is returned from `make_elf_from_assembly` (provided by pwntools)
+    """
+
+    qemu: subprocess.Popen = None
+
+    if QEMU_PORT is None:
+        print("'QEMU_PORT' environment variable not set")
+        sys.stdout.flush()
+        os._exit(1)
+
+    def _start_binary(asm: str, arch: str, *args):
+        nonlocal qemu
+
+        context.arch = arch
+        binary_tmp_path = make_elf_from_assembly(asm)
+
+        qemu = subprocess.Popen(
+            [
+                f"qemu-{arch}",
+                "-g",
+                f"{QEMU_PORT}",
+                f"{binary_tmp_path}",
+            ]
+        )
+
+        gdb.execute(f"target remote :{QEMU_PORT}")
+        gdb.execute("set exception-verbose on")
+
+        global _start_binary_called
+        # if _start_binary_called:
+        #     raise Exception('Starting more than one binary is not supported in pwndbg tests.')
+
+        _start_binary_called = True
```

Comment:
This lines (copied from gdb-tests fixtures) are likely remnants of an older testing system that ran multiple tests in the same GDB session. If this is the case, would it be best to remove them?

Reply:
Yes, lets remove them; can be in another PR since I will probably just hit merge on this if the code is good to go